### PR TITLE
added liquibase files and implemented soft delete logic

### PR DESCRIPTION
**Pull request by Den-moroz**

### CODE REVIEW

File: src/main/java/core/basesyntax/bookstore/controller/BookController.java
Comment (padu12) [inline src/main/java/core/basesyntax/bookstore/controller/BookController.java:1]: you just forgot to add the update endpoint and the corresponding method in the service class

**Review by padu12 — CHANGES_REQUESTED** [review]
Interesting solution. We have different implementations of the optional task and tbh I don't know which one is better🙃

And you forgot about some small thing)

**Review by padu12 — APPROVED** [review]
LGTM✨

File: src/main/java/core/basesyntax/bookstore/repository/book/BookSpecificationBuilder.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    private static final String KEY_FOR_TITLE = "title";
+    private static final String KEY_FOR_AUTHOR = "author";
+    private static final String KEY_FOR_PRICE_FROM = "fromPrice";
+    private static final String KEY_FOR_PRICE_TO = "toPrice";
+    private final SpecificationProviderManager<Book> bookSpecificationProviderManager;
+
+    @Override
+    public Specification<Book> build(BookSearchParametersDto searchParametersDto) {
+        Specification<Book> specification = Specification.where(null);
+        if (searchParametersDto.title() != null && searchParametersDto.title().length > 0) {
```
Comment (andrii-hoienko) [inline src/main/java/core/basesyntax/bookstore/repository/book/BookSpecificationBuilder.java]: code duplication, hint: in stringUntils there are methods wich perform logic you need here

**Review by andrii-hoienko — CHANGES_REQUESTED** [review]
very nice, only one small issue

**Review by Vasul-Pankiv — APPROVED** [review]

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 2]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-book-table.yaml
```
Comment (boroda4436) [inline src/main/resources/db/changelog/db.changelog-master.yaml:3]: empty line should be present at the EOF

File: src/main/java/core/basesyntax/bookstore/service/BookService.java [hunk 3]
```diff
@@ -10,4 +11,10 @@ public interface BookService {
     List<BookDto> findAll();
 
     BookDto getBookById(Long id);
+
+    BookDto updateById(Long id, CreateBookRequestDto bookRequestDto);
```
Comment (boroda4436) [inline src/main/java/core/basesyntax/bookstore/service/BookService.java:15]: could be `update` not `updateById`

**Pull request merged**